mapsforge-map/src/main/java/org/mapsforge/map/rendertheme/renderinstruction/RenderInstruction.java

```diff
@@ -173,4 +175,22 @@ protected Scale scaleFromValue(String value) {
      * @param scaleFactor the factor by which the text size should be scaled.
      */
     public abstract void scaleTextSize(float scaleFactor, byte zoomLevel);
+
+    /**
+     * Transform text before render based on the selected text transformation.
+     */
+    protected String transformText(String text, TextTransform transformation) {
+        if (text.length() == 0) {
+            return text;
+        }
+        if (transformation == TextTransform.UPPERCASE) {
+            return text.toUpperCase();
+        } else if (transformation == TextTransform.LOWERCASE) {
+            return text.toLowerCase();
+        } else if (transformation == TextTransform.CAPITALIZE) {
+            return text.substring(0, 1).toUpperCase() + text.substring(1);
```

Comment:
The case conversions are to be done with `Locale.getDefault()`?

Reply:
Yes, that's right. Do you think there is a better option?

Reply:
Some recommend `Locale.ROOT` in generic case conversions,
however these are local names, so not sure what is best option.

Reply:
Oki, thanks. Let's give it a try.